File: binds/go.v1/src/portalchell.go
package main

import (
	"fmt"
	"time"

	"gopkg.in/inconshreveable/log15.v2"
	"gopkg.in/project-iris/iris-go.v1"
)

func init() {
	iris.Log.SetHandler(log15.DiscardHandler())
}

// START OMIT
// Topic subscription handler processing inbound events
type chell struct{}
func (c *chell) HandleEvent(event []byte) { // HLsub
	fmt.Printf("%s\n\n", string(event))
}

func main() {
	// Connect to the Iris network as Chell
	connection, err := iris.Connect(55555) // HLsub
	if err != nil {
		fmt.Printf("Failed to connect: %v\n", err); return
	}
	defer connection.Close()

	fmt.Println("Tuning in to Aperture channels...")
	connection.Subscribe("official", new(chell), nil) // HLsub

	time.Sleep(time.Minute)
}
// END OMIT

File: binds/erlang.v1/src/portalchell.erl
-module(portalchell).
-export([main/0]).

%% START OMIT

-behavior(iris_topic). // HLsub
-export([init/1, handle_event/2, terminate/2]).

%% Implement the startup and cleanup methods of iris_topic.
init(nil)                  -> {ok, nil}. // HLsub
terminate(_Reason, _State) -> ok.        // HLsub

%% Print all events arriving on a subscription
handle_event(Event, State) -> // HLsub
    io:format("~s~n~n", [Event]),
    {noreply, State}.

main() ->
    {ok, Client} = iris_client:start(55555),

    io:format("Tuning in to Aperture channels...~n"),
    iris_client:subscribe(Client, "official", ?MODULE, nil), // HLsub

    timer:sleep(60 * 1000),
    ok = iris_client:stop(Client).
%% END OMIT
